libraries/botbuilder-ai/src/luisRecognizer.ts

```diff
@@ -417,40 +428,84 @@ export class LuisRecognizer implements LuisRecognizerTelemetryClient {
         telemetryMetrics?: Record<string, number>,
         options?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
+    ): Promise<RecognizerResult>;
+
+    /**
+     * Calls LUIS to recognize intents and entities in a users utterance.
+     *
+     * @param {string} utterance The utterance to be recognized.
+     * @param {LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions} options (Optional) options object used to override control predictions. Should conform to the [LuisRecognizerOptionsV2] or [LuisRecognizerOptionsV3] definition.
+     */
+    public async recognize(
+        utterance: string,
+        options?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
+    ): Promise<RecognizerResult>;
+
+    /**
+     * @internal
+     */
+    public async recognize(
+        contextOrUtterance: DialogContext | TurnContext | string,
+        maybeTelemetryPropertiesOrOptions?:
+            | Record<string, string>
+            | LuisRecognizerOptionsV2
+            | LuisRecognizerOptionsV3
+            | LuisPredictionOptions,
+        maybeTelemetryMetrics?: Record<string, number>,
+        maybeOptions?: LuisRecognizerOptionsV2 | LuisRecognizerOptionsV3 | LuisPredictionOptions
     ): Promise<RecognizerResult> {
-        const turnContext = context instanceof DialogContext ? context.context : context;
-        const cached = turnContext.turnState.get(this.cacheKey);
-        const luisRecognizer = options ? this.buildRecognizer(options) : this.luisRecognizerInternal;
-        if (!cached) {
-            const utterance = turnContext.activity.text || '';
-            let recognizerPromise: Promise<RecognizerResult>;
-
-            if (!utterance.trim()) {
-                // Bypass LUIS if the activity's text is null or whitespace
-                recognizerPromise = Promise.resolve({
-                    text: utterance,
-                    intents: { '': { score: 1 } },
-                    entities: {},
-                });
-            } else {
-                recognizerPromise = luisRecognizer.recognizeInternal(context);
-            }
+        // This type check, when true, logically implies that the function is being invoked as the two-argument string + optional options overload variant.
+        if (typeof contextOrUtterance === 'string') {
+            const utterance = contextOrUtterance;
+            const options = UnsafeLuisRecognizerUnion.optional().check(maybeTelemetryPropertiesOrOptions);
+
+            const luisRecognizer = options ? this.buildRecognizer(options) : this.luisRecognizerInternal;
+
+            return luisRecognizer.recognizeInternal(utterance);
+        } else {
+            const telemetryProperties = t
+                .Dictionary(t.String, t.String)
+                .optional()
+                .nullable()
+                .check(maybeTelemetryPropertiesOrOptions);
+            const turnContext =
```

Comment:
I wonder if we can get away from the `instanceof` by doing `const turnContext = contextOrUtterance.context ?? contextOrUtterance`. Does that compile?

Reply:
Unfortunately, not
![image](https://user-images.githubusercontent.com/40401643/118541811-00bd6b80-b707-11eb-95c4-07d95bff7e9f.png)